File: examples/javascript/constructs/block_shadowing.js
// A bare block at the top level binds names the program already has.
// The block's bindings last only inside it; after it, the outer ones are
// back. An assignment inside a block binds in the block too.
let n = 1;
let arr = [1, 2];
{
    let n = 20;
    let fresh = 5;
    console.log(n);
    console.log(fresh);
    arr.push(3);
}
console.log(n);
console.log(arr);
{
    n = 7;
    console.log(n);
}
console.log(n);
function inner() {
    let k = 1;
    {
        let k = 2;
        console.log(k);
    }
    {
        k = 3;
        console.log(k);
    }
    console.log(k);
}
inner();

File: examples/rust/constructs/block_shadowing.rs
// A bare block binds a name that an outer binding already has. The
// block's binding lives only inside the block; after it, the outer
// binding is back, and can still be read, written and grown.
fn main() {
    let mut v = [1, 2, 3];
    let mut n = 1;
    {
        let v = 10;
        let n = 20;
        println!("{} {}", v, n);
    }
    println!("{} {}", v, n);
    v.push(4);
    v[0] = 9;
    n = n + 1;
    println!("{} {} {}", v, v[0], n);
    {
        let n = 30;
        {
            let n = 40;
            println!("{}", n);
        }
        println!("{}", n);
    }
    println!("{}", n);
}

fn outer_seen_from_function() {
    let total = 0;
    {
        let total = 5;
        println!("{}", total);
    }
    println!("{}", total);
}
